src/SignalRServiceExtension/Config/DependencyInjectionExtensions.cs

```diff
@@ -0,0 +1,52 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using Microsoft.AspNetCore.SignalR.Protocol;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.DependencyInjection.Extensions;
+
+namespace Microsoft.Azure.WebJobs.Extensions.SignalRService
+{
+    internal static class DependencyInjectionExtensions
+    {
+        private const string HubProtocolError = "It's invalid to configure hub protocol on Azure Functions runtime V2. Newtonsoft.Json protocol will be used.";
+
+        public static IServiceCollection SetHubProtocol(this IServiceCollection services, IConfiguration configuration)
+        {
+            if (Environment.Version.Major == 4)
+            {
+                // Actually is .Net Core 2.x
+                if (configuration[Constants.AzureSignalRHubProtocol] != null)
+                {
+                    throw new InvalidOperationException(HubProtocolError);
+                }
+            }
+            else
+            {
+#if NETCOREAPP3_1
```

Comment:
missed 3.0 and 5.0+?

Reply:
Currently Azure Function only have 2.2 and 3.1 runtime.

Reply:
I see. So for 2.2 how about use NETCOREAPP2_2 instead of Environment major version looks tricky. Beside, can `#if` check can replace logical if check?

Reply:
Environment major version is more robust, because if uses set their target framework to netstandard 2.0 and run their function on netcoreapp2.2, then NETCOREAPP2_2 would not work.